orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/pipeline/persistence/ExecutionMapper.kt

```diff
@@ -132,4 +137,27 @@ class ExecutionMapper(
           }
       )
   }
+
+  @VisibleForTesting
+  fun convertPipelineRefTrigger(execution: PipelineExecution, context: DSLContext) {
```

Comment:
This is tied to the comment that although we are storing less data we are returning more data with this feature enabled. 

Maybe return a data object that matches the same data with enabled/disabled the feature?
In the future this will help us trim down the data returned only to what is necessary gaining more improvements.

Reply:
Basically the `stages` needs to be dropped from each parentExecution and this will match the previous returned object.

Reply:
I tested the behavior of orca in master and every time the SqlExecutionRepository retrieves an execution it contains all the stage's objects.
SqlExecutionRepository store the pipeline execution without stages because it split up the store for stages. In a few words the store is done in chunks (pipelines first, then stages) and for every update it is responsible on update pipeline executions & stage executions. But the retrieve always returns the full execution context including stages and all their data related.

If I drop the stage's data from a parentExecution it will lead to issues in other places in Orca because the SpEL expressions might fail because the data of the stages is not present and even in other places that I still don't know

I do think we need to improve more all the data retrieved for executions but that should be in another performance improvement.